### PR TITLE
browser: make reconnnect experience smoother

### PR DESCRIPTION
**Pull request by Rash419**
- don't invalidate the tiles
- don't reload the document if serverId changes
- writer: fix cursor jumps to top the document

Change-Id: Ie42850e77f9011cd1c820115919f10283407f216

* Resolves: # 
* Target version: master

### CODE REVIEW

File: browser/src/core/Socket.js [hunk 1]
```diff
@@ -1409,20 +1402,21 @@ app.definitions.Socket = L.Class.extend({
 		}
 		else if (this._reconnecting) {
 			// we are reconnecting ...
-			this._reconnecting = false;
-			this._map._docLayer._resetClientVisArea();
-			this._map._docLayer._requestNewTiles();
+			this._map._docLayer._refreshTilesInBackground();
 			this._map.fire('statusindicator', {statusType: 'reconnected'});
-			this._map._isNotebookbarLoadedOnCore = false;
```
Comment (eszkadev) [inline browser/src/core/Socket.js:1416]: This breaks notebookbar after reconnect. It requires to be inicialized properly in the core so our widgets like "style previews" will work.

File: browser/src/core/Socket.js [hunk 2]
```diff
@@ -1409,20 +1402,21 @@ app.definitions.Socket = L.Class.extend({
...
 			var uiMode = this._map.uiManager.getCurrentMode();
-			this._map.fire('changeuimode', {mode: uiMode, force: !window.migrating});
+			this._map.fire('changeuimode', {mode: uiMode, force: false});
```
Comment (eszkadev) [inline browser/src/core/Socket.js:1408]: we call `this.map.sendUnoCommand('.uno:ToolbarMode?Mode:string=notebookbar_online.ui');`  only onAdd currently in notebookbar. it has to be done here to be sure notebookbar's widgets work

**Comment by eszkadev** [comment]
for me it clears all the tiles (makes screen white) for a short time, so it could be more smooth :)

File: browser/src/core/Socket.js (hunk 1 above)
Comment (Rash419) [reply browser/src/core/Socket.js:1416]: Done thanks

**Review by eszkadev — CHANGES_REQUESTED** [review]
still notebookbar doesn't work after reconnect

File: browser/src/core/Socket.js [hunk 3]
```diff
@@ -1393,20 +1386,23 @@ app.definitions.Socket = L.Class.extend({
 		}
 		else if (this._reconnecting) {
 			// we are reconnecting ...
-			this._reconnecting = false;
```
Comment (eszkadev) [inline browser/src/core/Socket.js:1396]: if this `app.socket._reconnecting = false` is removed, where it is now done?

File: browser/src/core/Socket.js [hunk 4]
```diff
@@ -1393,20 +1386,23 @@ app.definitions.Socket = L.Class.extend({
...
-			this._map._docLayer._resetClientVisArea();
-			this._map._docLayer._requestNewTiles();
+			this._map._docLayer._refreshTilesInBackground();
 			this._map.fire('statusindicator', {statusType: 'reconnected'});
 			this._map._isNotebookbarLoadedOnCore = false;
 			var uiMode = this._map.uiManager.getCurrentMode();
-			this._map.fire('changeuimode', {mode: uiMode, force: !window.migrating});
+			if (uiMode === 'notebookbar') {
+				this._map.sendUnoCommand('.uno:ToolbarMode?Mode:string=notebookbar_online.ui');
```
Comment (eszkadev) [inline browser/src/core/Socket.js:1394]: this could be done via some helper function, I will try to do it later

**Review by eszkadev — APPROVED** [review]

**Comment by eszkadev** [comment]
now notebookbar issue is fixed

**Pull request merged**